pkgs/by-name/ra/rainbowcrack/package.nix

```diff
@@ -0,0 +1,37 @@
+{ lib
+, stdenvNoCC
+, fetchFromGitLab
+, alglib
+}:
+
+stdenvNoCC.mkDerivation rec {
+  pname = "rainbowcrack";
+  version = "1.8-0kali1";
+
+  src = fetchFromGitLab {
+    group = "kalilinux";
```

Comment:
is there a reason not to fetch from http://project-rainbowcrack.com/ ?

Reply:
3 weeks earlier i mentioned it here https://github.com/NixOS/nixpkgs/pull/290113#issuecomment-1996707280

Reply:
1.8 have issues during the unpackphase resulting in failed build but from the same source, 1.7 unpack successfully from the same source. so i chose Kali source as it packages 1.8 without issues.

Reply:
can you provide specifics on what is "failed build" ?

I had no issues upacking http://project-rainbowcrack.com/rainbowcrack-1.8-linux64.zip

also gentoo is using this same URL here
https://gitweb.gentoo.org/repo/gentoo.git/tree/app-crypt/rainbowcrack/rainbowcrack-1.8.ebuild

Reply:
maybe i am getting it wrong but when version 1.7 is used { lib
, stdenvNoCC
, fetchurl
, alglib
, unzip
}:

stdenvNoCC.mkDerivation rec {
  pname = "rainbowcrack";
  version = "1.7";

  src = fetchurl {
    url = "http://project-rainbowcrack.com/rainbowcrack-1.7-linux64.zip";
    hash = "sha256-iSaLYayVZpUyWEWjvim/psyJ3lrmKDNPWLe1gD5pE1M=";
  };

  buildInputs = [ alglib unzip ];

  installPhase = ''
    mkdir -p $out/{bin,share/rainbowcrack}
    cp *.txt $out/share/rainbowcrack
    cp -R rt* rcrack $out/bin
    chmod +x $out/bin/*
  '';

  meta = with lib; {
    description = "Rainbow table generator used for password cracking";
    homepage = "http://project-rainbowcrack.com";
    maintainers = with maintainers; [ tochiaha ];
    mainProgram = "rcrack";
  };
}

so when i change the version and the url to 1.8 and also the hash value, i end up with error. "Running phase: unpackPhase
unpacking source archive /nix/store/mqxjdz0ll4r0d9n8vvnqbljh6dbn55v3-rainbowcrack-1.8-linux64.zip
warning:  /nix/store/mqxjdz0ll4r0d9n8vvnqbljh6dbn55v3-rainbowcrack-1.8-linux64.zip appears to use backslashes as path separators
do not know how to unpack source archive /nix/store/mqxjdz0ll4r0d9n8vvnqbljh6dbn55v3-rainbowcrack-1.8-linux64.zip
error: builder for '/nix/store/z5p57zp0l6bv983c0mz19gjga7yz73kh-rainbowcrack-1.8.drv' failed with exit code 1"

Reply:
ok, i see the issue now

it can be worked around by

```
  unpackPhase = ''
    ${pkgs.unzip}/bin/unzip $src || true
  '';

```

but I will leave it up to you if you want to continue to use kali repo

Reply:
Since it has received more comments about fetching from source, i will make changes to fetch package from source.

Reply:
fetching from kali is fine, just need to leave a comment explaining why. because its unusual that why people ask